python/summarize_sgfs.py

```diff
@@ -0,0 +1,295 @@
+import itertools
+
+from sgfmill import sgf
+import elo
+import os
+import math
+import argparse
+from typing import List, Dict, Tuple, Set, Sequence
+from dataclasses import dataclass
+import warnings
+
+
+@dataclass
+class Record:
+  win: int = 0
+  lost: int = 0
+  draw: int = 0
+
+
+class GameResultSummary:
+  """
+  Summrize Go games results in sgf or sgfs file format under a directory (and its subdirectory).
+  The sgfs file is a special file format provided by Katago, and each row is a single game in a sgf string.
+  sgf file is not required to be in a single line.
+
+  Public functions include:
+  self.add_games(input_file_dir, search_subdir = False): add files into the results and duplicate files will not be
+  added.
+  self.add_a_game_file(input_file_name): add a new game file into the results and duplicate file will not be added.
+  self.clear(): clear all the results
+  self.print_elos(): print the estimated Bayes Elo scores
+  self.print_game_results(): print the game results
+  self.get_game_results(self) -> Dict
+
+  Example:
+    call it from terminal:
+      :$python summarize_sgfs.py -input-dirs [input_directory1 input directory2] -elo-prior 3400
+
+    call it by other function:
+    import summarize_sgfs
+    elo_prior = 3400
+    gameResultSummary = summarize_sgfs.GameResultSummary(elo_prior)
+    gameResultSummary.add_games(input_file_dir)
+    gameResultSummary.print_game_results()
+    gameResultSummary.print_elos()
+  """
+
+  def __init__(
+      self,
+      elo_prior: float
+  ):
+    self.results = {}  # a dictionary as {(black_player_name, white_player_name):Record}
+
+    self._all_sgfs_files = set()
+    self._all_sgf_files = set()
+    self._new_sgfs_files = set()
+    self._new_sgf_files = set()
+    self._warning = False
```

Comment:
Very minor: `_warning` could be better named `_should_warn_handicap_komi` or something like that.

Reply:
Agreed. Changed.